sound/soc/sof/topology.c

```diff
@@ -1752,15 +1752,15 @@ static int sof_widget_load_siggen(struct snd_soc_component *scomp, int index,
 	return ret;
 }
 
-static size_t sof_get_control_data(struct snd_sof_dev *sdev,
-				   struct snd_soc_dapm_widget *widget,
-				   struct sof_widget_data *wdata)
+static int sof_get_control_data(struct snd_sof_dev *sdev,
+				struct snd_soc_dapm_widget *widget,
+				struct sof_widget_data *wdata,
+				size_t *size)
```

Comment:
maybe safer to initialize *size = 0 here, instead of the calling function?

Also you want to change the commit message to add the 'Fixes' and 'Reported-by' tags to give Dan Carpenter credit for his review work.